### PR TITLE
Create Cobra taskrunner for backfilling

### PR DESCRIPTION
**Pull request by edisonz0718**
This PR:
1. Create a cobra app in bastion service for running operation/admin tasks
2. Create the backfill app and backfill the nft data
3. Move common pkg outside of app folder
4. Update create NFT handler to createOrUpdate nft.

Backfill in dev cluster:
Screen Shot 2023-02-17 at 11 27 40 PM

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -115,4 +115,4 @@ lint:
 
 .PHONY: abigen
 abigen:
-	cd api; mkdir ./internal/abi/${package}; abigen --abi=./resource/abi/${package}.json --pkg=${package} --out=./internal/abi/${package}/${package}.go
\ No newline at end of file
+	mkdir ./pkg/abi/${package}; abigen --abi=./resource/abi/${package}.json --pkg=${package} --out=./pkg/abi/${package}/${package}.go
```
Comment (LeoHChen) [inline Makefile:118]: do you want to use `mkdir -p` to ignore the error of existing directories?

File: api/internal/handler/nft_handler.go [hunk 2]
```diff
@@ -171,7 +188,7 @@ func createNftRecord(uri string, tokenId int, ownerAddress string, collectionAdd
 		return nil, fmt.Errorf("failed to split uri string to 2 parts. uri: %v", uri)
 	}
 
-	decodedMetadata, err := base64.URLEncoding.DecodeString(splittedStr[1])
+	decodedMetadata, err := base64.StdEncoding.DecodeString(splittedStr[1])
```
Comment (LeoHChen) [inline api/internal/handler/nft_handler.go:191]: why this change?

File: bastion/task/cmd/backfill.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
+/*
+Copyright © 2023 NAME HERE <EMAIL ADDRESS>
```
Comment (LeoHChen) [inline bastion/task/cmd/backfill.go:2]: ?

File: bastion/task/cmd/backfill.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+var backfillCmd = &cobra.Command{
+	Use:   "backfill",
+	Short: "backfill task for backfilling nft onchain data to database",
+	Long: `The task accepts flags: --start, ---end, --collection.
+--collection is required and will be used to determine which nft collection to backfill.
+--start and --end tell the task the range of nft token id to backfill. When they are specified. The default is to backfill all nft data from a colleciton.	
+	`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if collectionAddress == "" {
+			fmt.Printf("[ERROR] collection address required. pass it with the flag --collection")
```
Comment (LeoHChen) [inline bastion/task/cmd/backfill.go:33]: do we have a common logging module?  Or you may create a github issue?  we used to use zerolog in previous project.

File: api/internal/repository/nft_token.go [hunk 5]
```diff
@@ -151,6 +152,24 @@ func (n *nftTokenDbImpl) CreateNft(nftToken *NftTokenModel) (*NftTokenModel, err
 	return nftToken, nil
 }
 
+func (n *nftTokenDbImpl) UpdateNft(nftToken *NftTokenModel) (*NftTokenModel, error) {
+	nftToken.CollectionAddress = strings.ToLower(nftToken.CollectionAddress)
```
Comment (LeoHChen) [inline api/internal/repository/nft_token.go:156]: seems like you may need a generic function to check and convert the addresses, may create an issue.

File: bastion/config/config.go [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+	configPath = "/bastion/config"
+)
+
+// GetConfig loads the config
+func GetConfig() (*TaskConfig, error) {
+	var cfg TaskConfig
+
+	path, err := os.Getwd()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get config directories")
```
Comment (LeoHChen) [inline bastion/config/config.go:30]: why not logger here?

File: bastion/config/local.yaml [hunk 7]
```diff
@@ -0,0 +1,2 @@
+provider_url: "https://eth-goerli.g.alchemy.com/v2/RWhchAQNylFZLnnPO7Rmj3b4T4uZIFuO"
```
Comment (LeoHChen) [inline bastion/config/local.yaml:1]: is this for local test or dev test?

File: Makefile (hunk 1 above)
Comment (edisonz0718) [reply Makefile]: Yes. Thanks for the suggestion.

File: api/internal/handler/nft_handler.go (hunk 2 above)
Comment (edisonz0718) [reply api/internal/handler/nft_handler.go:191]: The URLEncoding failed to decode some of the URIs in the NFT contract. But StdEncoding works for all tokens.

File: bastion/task/cmd/backfill.go (hunk 3 above)
Comment (edisonz0718) [reply bastion/task/cmd/backfill.go]: This was auto generated by cobra. Deleted this section for all the cobra generated files.

File: bastion/task/cmd/backfill.go (hunk 4 above)
Comment (edisonz0718) [reply bastion/task/cmd/backfill.go]: Yes. I use zap from Uber. Didn't add logger here because I thought, for operation scripts, probably not much benefits to add it.  But I just added to make it consistent.

File: bastion/config/local.yaml (hunk 7 above)
Comment (edisonz0718) [reply bastion/config/local.yaml:1]: Forgot to put it in gitignore since I added later. Just added it to gitignore.

File: bastion/config/config.go (hunk 6 above)
Comment (edisonz0718) [reply bastion/config/config.go:30]: fmt.Errorf is to return an error with type 'error'.

File: api/internal/repository/nft_token.go (hunk 5 above)
Comment (edisonz0718) [reply api/internal/repository/nft_token.go:156]: Yes. Created an issue.

**Review by LeoHChen — APPROVED** [review]
LGTM, but you have to address the conflicts now after I rebased/merged your previous PR.
